ir/memory.cpp

```diff
@@ -1512,7 +1516,8 @@ Memory::load(const expr &p, const Type &type, unsigned align) {
   auto ubs = ptr.is_dereferenceable(bytecount, align, false);
 
   StateValue ret;
-  if (auto aty = type.getAsAggregateType()) {
+  auto aty = type.getAsAggregateType();
+  if (aty && !isVectorWithNonByteInts(type)) {
```

Comment:
can we use the code for all vectors (not just non-byte aligned)?
This would allow to widen all vector load/stores.

Reply:
I guess non-ptr vectors only.

Reply:
Updated